docs/contributing.md

```diff
@@ -0,0 +1,48 @@
+# Contributing to drone-helm3
+
+We're glad you're interested in contributing! Here are some guidelines that will help make sure everyone has a good experience:
+
+## Submitting a patch
+
+Before you start working on a change, please make sure there's an associated issue. It doesn't need to be thoroughly scrutinized and dissected, but it needs to exist.
+
+Please put the relevant issue number in the first line of your commit messages, e.g. `vorpalize the frabjulator [#42]`. Branch names do not need issue numbers, but feel free to include them if you like.
+
+We encourage you to follow [the guidelines in Pro Git](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines) when making commits. In short:
+
+* Commit early and commit often.
+* Make the first line of the commit message concise--no more than 50 characters or so.
+* Make the rest of the commit message verbose--information about _why_ you did what you did is particularly helpful.
+
+Once you're satisfied with your work, send us a pull request. If you'd like, you can send the pull request _before_ you're satisfied with your work; just be sure to mark the PR a draft or put `[WIP]` in the title.
+
+## How to run the tests
+
+We use `go test`, `go vet`, and `golint`:
+
+```
+go test ./cmd/... ./internal/...
+go vet ./cmd/... ./internal/...
+golint -set_exit_status ./cmd/... ./internal/...
+```
+
+If you have [the Drone cli tool](https://docs.drone.io/cli/install/) installed, you can also use `drone exec --include test --include lint`.
+
+## Using the plugin locally
```

Comment:
Following these instructions is pretty painful. Are they worth including? Is there a better way to do it?

Reply:
Are you talking about using environment variables? Probably an other way would be to provide a sample ".env" file so it's a little easier, but I'm not sure what else would make it less painful

Reply:
Mostly just the initial setup—you need to stand up a docker registry, stand up a kube cluster (I never did get one working locally), and find a fixture repo. 

`.env` would resolve the "uncomment stuff in `.drone.yml`" part, but you can't just put `DOCKER_USERNAME` and `DOCKER_PASSWORD` in your .env until you've configured basic auth for the registry image, and it won't let you configure basic auth unless you configure TLS first, so standing up the registry image is even more complicated.

Maybe it would work to put `insecure: {from_secret: docker_connect_insecurely}` in the real `publish_linux_amd64` step, then advise setting that to true in `.env`. I'm not sure how the publish plugin behaves if `insecure` and credentials are both present.

Reply:
Why do you need a docker registry? - kube cluster needs to be there period so anyone will need to have one of those setup.

Reply:
When you run a `drone exec` for the fixture repo, that repo's `.drone.yml` needs to be able to find the new plugin image. I guess you could just create an account on dockerhub and make a "this is a dumping ground for testing" repository, though. That would make it a lot simpler.

Reply:
Regarding the kube cluster: having the cluster is easy, but configuring it to accept token-based auth is complicated. I actually never managed it, just switched to using our Rancher cluster.

Reply:
I think there's a couple things here.
1. if we're trying to "test" everything in drone, I would run the test from the binary created and not the "container". Something like, 1-unit test, 2-integration test (although this would still require k8s) in that case I'd look at running k8s in docker as a drone service (I think that's possible but not 100% on it) although I like the idea of that long term, probably overkill for now
2. make it so that drone exec only runs unit test and builds the binary locally. then you can manually build the dockerimage locally and run it that way (that's how I've done most of the plugins up to this point and probably a good way to go until more people use this one)